calling-apis/managing-asynchronous-calls.md

```diff
@@ -0,0 +1,277 @@
+- Finds the coordinates of Seattle by calling `findLatitudeAndLongitude` and puts it in `seattleCoordinates`
+- Finds the location results by calling `findLocations` and puts it in `locations`
+
+However, when we run this code, our console outputs this:
+
+```
+error in findLocation!
+success in findLatitudeAndLongitude! 47.6038321 -122.3300624
+```
+
+Raffy's experiment doesn't look successful at all! What's noticeable is that `error in findLocation!` prints out **_before_** `success in findLatitudeAndLongitude!` does.
+
+It's apparent that the API call in `findLatitudeAndLongitude` works. We can get back a latitude and longitude just fine!
+
+Why would there be an error when calling `findLocation`? And why would this print **_before_** we see a print statement from `findLatitudeAndLongitude`?
+
+<!-- Question 1 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: e723527e
+* title: Managing Asynchronous Calls
+##### !question
+
+Make a hypothesis:
+
+Why would there be an error when calling `findLocation`? And why would this print **_before_** we see a print statement from `findLatitudeAndLongitude`?
+
+##### !end-question
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+### Debugging Our Problem
+
+After debugging, Raffy should hone in on one specific line:
+
+```js
+const locations = findLocation(seattleCoordinates.seattleLat, seattleCoordinates.seattleLon);
+```
+
+Raffy found these two facts while debugging:
+
+- `seattleCoordinates.seattleLat` _and_ `seattleCoordinates.seattleLon` are `undefined` at the time of calling `findLocation`
+- The print statements in `findLocation` printed, meaning that `findLocation` was definitely called and got into the `catch` block
+
+After diagramming, thinking, researching, and rubber-ducking, Raffy comes to this conclusion:
+
+- `seattleCoordinates` has undefined values because it hasn't been assigned a value yet, and JavaScript gives `undefined` when accessing `seattleCoordinates.seattleLat` and `seattleCoordinates.seattleLon`
+- `seattleCoordinates` is assigned a value from `findLatitudeAndLongitude`'s return
+- `seattleCoordinates` wasn't assigned a value because `findLatitudeAndLongitude` didn't return yet
+- Something that changes the flow of control in our code is our asynchronous API calls
+
+### Refactoring: Utilize `then`
+
+How can Raffy ensure that `findLocation` executes after `findLatitudeAndLongitude` finishes?
+
+Raffy has a few options:
+
+- Re-order the code, so that the second API call happens **inside** the `then` chained to the first API call
+- Depending on our depth of knowledge in JavaScript, approach this from a different design pattern
+- Depending on different tools/libraries available, approach this from a different design pattern
+
+When working on a smaller JavaScript project, we can approach solving this problem by refactoring our work into the first `then` block.
```

Comment:
for the two "depending" statements, I wouldn't say "approach from a different design pattern". Rather "approach from a different perspective/direction/angle", or "approach using a different pattern". For good measure, I used both variations.

Explicitly say this approach is just one way of solving this, and be more explicit about what our approach will be.

```suggestion
- Depending on our depth of knowledge in JavaScript, approach this from a different design perspective
- Depending on the tools/libraries available, approach this using a different design pattern

When working on a smaller JavaScript project, one way we can approach solving this problem is by refactoring our second API call into a `then` block of the first API call.
```